pkg/resource/resource.go

```diff
@@ -33,136 +33,74 @@ type resourceCallbacks map[string]func(obj *unstructured.Unstructured, sr interf
 
 var (
 	customCallback = make(resourceCallbacks)
-	log            = zap.New(zap.UseDevMode(true)).WithName(color.Print("resource", color.Blue))
 	HelmClient     kube.Interface
 	RuntimeScheme  *runtime.Scheme
 	UpdateVendor   string
 	pollActions    = poll.New()
-	metricsClient  = metrics.New()
 )
 
-func IsNamespaced(kind string) bool {
-	if kind == "Namespace" ||
-		kind == "ClusterRole" ||
-		kind == "ClusterRoleBinding" ||
-		kind == "SecurityContextConstraint" ||
-		kind == "SpecialResource" {
-		return false
-	}
-	return true
+type Creator interface {
+	CreateFromYAML([]byte, bool, v1.Object, string, string, map[string]string, string, string) error
 }
 
-func IsNotUpdateable(kind string) bool {
-	// ServiceAccounts cannot be updated, maybe delete and create?
-	if kind == "ServiceAccount" || kind == "Pod" {
-		return true
-	}
-	return false
+type creator struct {
+	kubeClient    clients.ClientsInterface
+	log           logr.Logger
+	metricsClient metrics.Metrics
 }
 
-// Some resources need an updated resourceversion, during updates
-func NeedsResourceVersionUpdate(kind string) bool {
-	if kind == "SecurityContextConstraints" ||
-		kind == "Service" ||
-		kind == "ServiceMonitor" ||
-		kind == "Route" ||
-		kind == "Build" ||
-		kind == "BuildRun" ||
-		kind == "BuildConfig" ||
-		kind == "ImageStream" ||
-		kind == "PrometheusRule" ||
-		kind == "CSIDriver" ||
-		kind == "Issuer" ||
-		kind == "CustomResourceDefinition" ||
-		kind == "Certificate" ||
-		kind == "SpecialResource" ||
-		kind == "OperatorGroup" ||
-		kind == "CertManager" ||
-		kind == "MutatingWebhookConfiguration" ||
-		kind == "ValidatingWebhookConfiguration" ||
-		kind == "Deployment" ||
-		kind == "ImagePolicy" {
-		return true
+func NewCreator(metricsClient metrics.Metrics, kubeClient clients.ClientsInterface) Creator {
+	return &creator{
+		metricsClient: metricsClient,
+		log:           zap.New(zap.UseDevMode(true)).WithName(color.Print("resource", color.Blue)),
+		kubeClient:    kubeClient,
 	}
-	return false
-
 }
 
-func UpdateResourceVersion(req *unstructured.Unstructured, found *unstructured.Unstructured) error {
-
-	kind := found.GetKind()
+func (c *creator) AfterCRUD(obj *unstructured.Unstructured, namespace string) error {
```

Comment:
This function is used only in resource package, lets make it private

Reply:
We agreed to address this with https://issues.redhat.com/browse/MGMT-8567